homeassistant/components/velux/config_flow.py

```diff
@@ -43,13 +54,54 @@ async def async_step_user(
                 errors["base"] = "unknown"
             else:
                 return self.async_create_entry(
-                    title=user_input[CONF_HOST],
+                    title=user_input[CONF_NAME] or user_input[CONF_HOST],
                     data=user_input,
                 )
 
-        data_schema = self.add_suggested_values_to_schema(DATA_SCHEMA, user_input)
+        data_schema = self.discovery_schema or self.add_suggested_values_to_schema(
+            DATA_SCHEMA, user_input
+        )
         return self.async_show_form(
             step_id="user",
             data_schema=data_schema,
             errors=errors,
         )
+
+    async def async_step_dhcp(
+        self, discovery_info: DhcpServiceInfo
+    ) -> ConfigFlowResult:
+        """Handle discovery by DHCP."""
+        return await self._async_handle_discovery(
+            {
+                CONF_HOST: discovery_info.ip,
+                CONF_MAC: format_mac(discovery_info.macaddress),
+                CONF_NAME: discovery_info.hostname.upper().replace("LAN_", ""),
+            }
+        )
+
+    async def _async_handle_discovery(
+        self, discovery_info: dict[str, Any]
+    ) -> ConfigFlowResult:
+        """Prepare configuration for a discovered Velux device."""
+        # The hostname ends with the last 4 digits of the device MAC address.
+        await self.async_set_unique_id(discovery_info[CONF_NAME])
```

Comment:
I think we come back to the point that the current entries don't have an unique id, so merging this PR will cause everyone to see a discovered entry and I don't know if that is what we want

Reply:
see below